crypto/trace.c

```diff
@@ -126,8 +126,6 @@ static const struct trace_category_st trace_categories[] = {
     TRACE_CATEGORY_(TLS),
     TRACE_CATEGORY_(TLS_CIPHER),
     TRACE_CATEGORY_(CONF),
-    TRACE_CATEGORY_(ENGINE_TABLE),
-    TRACE_CATEGORY_(ENGINE_REF_COUNT),
```

Comment:
Hmmm, shouldn't this rather be wrapped in a `#ifndef OPENSSL_NO_ENGINE`?

Reply:
Do we really need to support engine tracing?  This is a new feature and engines are being deprecated.  No engine user will miss this.

Note also: the documentation mentions _OSSL_TRACE_CATEGORY_ENGINE_CONF_ which isn't defined anywhere.  I've added a fixup to remove this too.

Reply:
I've added the conditionals.

Also removed the _OSSL_TRACE_CATEGORY_ENGINE_CONF_ documentation.